[github] apiEndpoint is not considered and this cause problem with Github enterprise

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [ ] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Configure host and apiEndpoint with your GitHub enterprise 
2. do a jreleaser release

### Expected Behaviour

Release should be pusblished

### Actual Behaviour
apiEndpoint is not used. 

![image](https://user-images.githubusercontent.com/2572323/136575686-97abef02-af2f-4352-864a-9ee7d0eb9568.png)

Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://**api.github.com**/repos/<omitted_org>/test-repo
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1924)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:527)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:334)
	at org.kohsuke.github.GitHubHttpUrlConnectionClient.getResponseInfo(GitHubHttpUrlConnectionClient.java:63)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:387)
	... 26 more

### Environment Information

- **Operating System**: Windows
- **JReleaser Version:** 0.7.0
- **JDK Version:** 11

--- a/sdks/github-java-sdk/src/main/java/org/jreleaser/sdk/github/GithubReleaser.java
+++ b/sdks/github-java-sdk/src/main/java/org/jreleaser/sdk/github/GithubReleaser.java
@@ -64,6 +64,7 @@ public class GithubReleaser extends AbstractReleaser {
             String changelog = context.getChangelog();
 
             Github api = new Github(context.getLogger(),
+                github.getApiEndpoint(),
                 github.getResolvedToken(),
                 github.getConnectTimeout(),
                 github.getReadTimeout());

